src/main/java/com/google/api/codegen/util/ConfigNextVersionValidator.java

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import com.google.protobuf.util.JsonFormat.Parser;
+import com.google.protobuf.util.JsonFormat.Printer;
+import com.google.protobuf.util.JsonFormat.TypeRegistry;
+import javax.annotation.Nonnull;
+
+public class ConfigNextVersionValidator {
+
+  public static String CONFIG_V2_MAJOR_VERSION = "2";
+  public static String CONFIG_V2_VERSION = CONFIG_V2_MAJOR_VERSION + ".0.0"; // "2.0.0"
+
+  /**
+   * Throw {@link IllegalStateException} iff the given input contains fields unknown to the {@link
+   * com.google.api.codegen.v2.ConfigProto} schema.
+   */
+  public void checkIsNextVersionConfig(@Nonnull com.google.api.codegen.ConfigProto configV1Proto)
```

Comment:
What about something like `validateV2Config`? I don't think it is desirable to make the function name version-agnostic (unless the implementation is also version-agnostic)

Reply:
ok!